### PR TITLE
trivial: Add fu_common_uri_get_scheme() for future use

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

File: libfwupdplugin/fu-common.c [hunk 1]
```diff
@@ -2854,3 +2854,26 @@ fu_common_crc32 (const guint8 *buf, gsize bufsz)
...
+ **/
+gchar *
+fu_common_uri_get_scheme (const gchar *uri)
+{
+	gchar *tmp;
+
+	g_return_val_if_fail (uri != NULL, NULL);
+
+	tmp = g_strstr_len (uri, -1, ":");
+	if (tmp == NULL || tmp[0] == '\0')
```
Comment (superm1) [inline libfwupdplugin/fu-common.c:2876]: as it's a pointer that is returned, shouldn't it be `tmp == NULL || *tmp == '\0'`
(i'll admit functionality equivalent)

File: libfwupdplugin/fu-common.c [hunk 2]
```diff
@@ -2854,3 +2854,26 @@ fu_common_crc32 (const guint8 *buf, gsize bufsz)
...
+		return NULL;
+	return g_strndup (uri, tmp - uri);
```
Comment (superm1) [inline libfwupdplugin/fu-common.c:2878]: I think you want to have this always return lowercase

File: libfwupdplugin/fu-self-test.c [hunk 3]
```diff
@@ -350,6 +350,26 @@ fu_common_strsafe_func (void)
...
+fu_common_uri_scheme_func (void)
+{
+	struct {
+		const gchar *in;
+		const gchar *op;
+	} strs[] = {
+		{ "https://foo.bar/baz",	"https" },
+		{ "ftp://",			"ftp" },
+		{ "ftp:",			"ftp" },
+		{ "foobarbaz",			NULL },
```
Comment (superm1) [inline libfwupdplugin/fu-self-test.c:363]: I think you need a case insensitive test

File: libfwupdplugin/fu-common.c (hunk 1 above)
Comment (hughsie) [reply libfwupdplugin/fu-common.c:2876]: it's technically a `gchar *` pointer, not a gpointer.

File: libfwupdplugin/fu-common.c (hunk 2 above)
Comment (hughsie) [reply libfwupdplugin/fu-common.c:2878]: Good catch, agree.

File: libfwupdplugin/fu-self-test.c (hunk 3 above)
Comment (hughsie) [reply libfwupdplugin/fu-self-test.c:364]: Fixed.

**Review by superm1 — APPROVED** [review]

**Pull request merged**